c_emulator/rvfi_dii.cpp

```diff
@@ -88,40 +221,25 @@ bool rvfi_handler::setup_socket(bool config_print)
   return true;
 }
 
-void rvfi_handler::get_and_send_packet(packet_reader_fn reader,
-                                       bool config_print)
+template <typename T>
+void rvfi_handler::send_packet(const T *data, bool config_print)
```

Comment:
```suggestion
void rvfi_handler::send_packet(const T &data, bool config_print)
```
Since we are writing only one element I would use a reference to show that it's not an array.